File: src/main/java/org/fb/manifest/ui/FxStage.java
package org.fb.manifest.ui;

import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;
import org.fb.manifest.StageReadyEvent;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationListener;
import org.springframework.core.io.ClassPathResource;
import org.springframework.stereotype.Component;

@Component
class FxStage implements ApplicationListener<StageReadyEvent> {
    private final ApplicationContext context;
    private Stage stage;

    FxStage(ApplicationContext context) {
        this.context = context;
    }

    public void onApplicationEvent(StageReadyEvent event) {
        try {

            stage = event.getStage();
            FXMLLoader fxmlLoader = new FXMLLoader(new ClassPathResource("manifest.fxml").getURL());
            fxmlLoader.setControllerFactory(param -> context.getBean(param));

            Parent parent = (Parent) fxmlLoader.load();
            stage.setScene(new Scene(parent));
            stage.setTitle("Folder diff");
            stage.show();

        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/kotlin/org/fb/kdiff/ui/FxStage.kt
package org.fb.kdiff.ui

import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.stage.Stage
import javafx.util.Callback
import org.fb.kdiff.StageReadyEvent
import org.springframework.context.ApplicationContext
import org.springframework.context.ApplicationListener
import org.springframework.core.io.ClassPathResource
import org.springframework.stereotype.Component

@Component
class FxStage(private val context: ApplicationContext) : ApplicationListener<StageReadyEvent> {

    private lateinit var stage: Stage

    override fun onApplicationEvent(event: StageReadyEvent) = try {

        stage = event.stage
        val fxmlLoader = FXMLLoader(ClassPathResource("kdiff.fxml").url)
        fxmlLoader.controllerFactory = Callback<Class<*>, Any> { context.getBean(it) }

        val parent = fxmlLoader.load<Parent>()
        stage.scene = Scene(parent)
        stage.title = "Folder diff"
        stage.show()

    } catch (e: Exception) {
        throw RuntimeException(e)
    }
}